### PR TITLE
refactor ui_draw_driver_view

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/ui/paint.cc [hunk 1]
```diff
@@ -371,56 +371,40 @@ static void ui_draw_vision_face(UIState *s) {
...
-    if (std::abs(face_x) <= 0.35 && std::abs(face_y) <= 0.4) {
-      ui_draw_rect(s->vg, fbox_x, fbox_y, 0.6 * box_h / 2, 0.6 * box_h / 2,
-                   nvgRGBAf(1.0, 1.0, 1.0, 0.8 - ((std::abs(face_x) > std::abs(face_y) ? std::abs(face_x) : std::abs(face_y))) * 0.6 / 0.375),
-                   35, 10);
-    } else {
-      ui_draw_rect(s->vg, fbox_x, fbox_y, 0.6 * box_h / 2, 0.6 * box_h / 2, nvgRGBAf(1.0, 1.0, 1.0, 0.2), 35, 10);
-    }
+  const int blackout_x = is_rhd ? rect.x : valid_rect.right();
+  const int blackout_w = rect.w - valid_rect.w;
+  NVGpaint gradient = nvgLinearGradient(s->vg, blackout_x, rect.y, blackout_x + blackout_w, rect.y,
```
Comment (adeebshihadeh) [inline selfdrive/ui/paint.cc:384]: there's a small gap between the edge of the frame and the gradient now

File: selfdrive/ui/paint.cc [hunk 2]
```diff
@@ -371,56 +371,40 @@ static void ui_draw_vision_face(UIState *s) {
...
+                                        COLOR_BLACK_ALPHA(is_rhd ? 255 : 0), COLOR_BLACK_ALPHA(is_rhd ? 0 : 255));
+  ui_draw_rect(s->vg, blackout_x, rect.y, blackout_w, rect.h, gradient);
+  ui_draw_rect(s->vg, blackout_x, rect.y, blackout_w, rect.h, COLOR_BLACK_ALPHA(144));
+
+  const bool face_detected = s->scene.dmonitoring_state.getFaceDetected();
+  if (face_detected) {
+    auto fxy_list = s->scene.driver_state.getFacePosition();
+    auto [face_x, face_y] = (float[]){fxy_list[0], fxy_list[1]};
```
Comment (adeebshihadeh) [inline selfdrive/ui/paint.cc:392]: I think this was cleaner the way it was. we can afford the extra line :)

File: selfdrive/ui/paint.cc [hunk 3]
```diff
@@ -371,56 +371,40 @@ static void ui_draw_vision_face(UIState *s) {
...
+    ui_draw_rect(s->vg, fbox_x - box_size / 2, fbox_y - box_size / 2, box_size, box_size, nvgRGBAf(1.0, 1.0, 1.0, alpha), 35, 10);
   }
 
   // draw face icon
   const int face_size = 85;
-  const int x = (valid_frame_x + face_size + (bdr_s * 2)) + (scene->is_rhd ? valid_frame_w - box_h / 2:0);
-  const int y = (box_y + box_h - face_size - bdr_s - (bdr_s * 1.5));
-  ui_draw_circle_image(s->vg, x, y, face_size, s->img_face, scene->dmonitoring_state.getFaceDetected());
+  const int icon_x = is_rhd ? rect.right() - face_size - bdr_s * 2 : rect.x + face_size + bdr_s * 2;
+  ui_draw_circle_image(s->vg, icon_x, rect.bottom() - face_size - bdr_s * 2.5, face_size, s->img_face, face_detected);
```
Comment (adeebshihadeh) [inline selfdrive/ui/paint.cc:407]: same here, give the y pos its own line

**Comment by deanlee** [comment]
I used draw_rect to fix this gap.

That one pixel gap has always existed before,it was covered by draw_rect.maybe we need to find out why this happens next time.

**Pull request merged**